components/helm-broker/internal/controller/addonsconfiguration_controller.go

```diff
@@ -0,0 +1,500 @@
+			for _, add := range repo.Addons {
+				if add.Status == addonsv1alpha1.AddonStatusReady {
+
+					r.log.Infof("- delete DocsTopic for bundle %s", add)
+					b, err := r.bundleStorage.Get(internal.Namespace(addon.Namespace), internal.BundleName(add.Name), *semver.MustParse(add.Version))
+					if err != nil {
+						return exerr.Wrapf(err, "while getting bundle %s from namespace %s", add.Name, addon.Namespace)
+					}
+					if err := r.docsTopicProvider.EnsureDocsTopicRemoved(string(b.ID), addon.Namespace); err != nil {
+						return exerr.Wrapf(err, "while ensuring ClusterDocsTopic for bundle %s is removed", b.ID)
+					}
+
+					r.log.Infof("- delete bundle %s from namespace %s", add, addon.Namespace)
+					if err := r.bundleStorage.Remove(internal.Namespace(addon.Namespace), internal.BundleName(add.Name), *semver.MustParse(add.Version)); err != nil {
+						return exerr.Wrapf(err, "while deleting bundle %s from storage", add.Name)
+					}
+				}
+			}
+		}
+	}
+
+	if err := r.deleteFinalizer(addon); err != nil {
+		return exerr.Wrapf(err, "while deleting finalizer for AddonConfiguration %s/%s", addon.Name, addon.Namespace)
+	}
+
+	r.log.Info("Delete AddonsConfiguration process completed")
+	return nil
+}
+
+func (r *ReconcileAddonsConfiguration) existingAddonsConfigurationList(addon *addonsv1alpha1.AddonsConfiguration) (*addonsv1alpha1.AddonsConfigurationList, error) {
+	addonsList := &addonsv1alpha1.AddonsConfigurationList{}
+	addonsConfigurationList, err := r.addonsConfigurationList(addon.Namespace)
+	if err != nil {
+		return nil, exerr.Wrapf(err, "while listing AddonsConfigurations from namespace %s", addon.Namespace)
+	}
+
+	for _, existAddon := range addonsConfigurationList.Items {
+		if existAddon.Name != addon.Name {
+			addonsList.Items = append(addonsList.Items, existAddon)
+		}
+	}
+
+	return addonsList, nil
+}
+
+func (r *ReconcileAddonsConfiguration) addonsConfigurationList(namespace string) (*addonsv1alpha1.AddonsConfigurationList, error) {
+	addonsConfigurationList := &addonsv1alpha1.AddonsConfigurationList{}
+
+	err := r.Client.List(context.TODO(), &client.ListOptions{Namespace: namespace}, addonsConfigurationList)
+	if err != nil {
+		return addonsConfigurationList, exerr.Wrap(err, "during fetching AddonConfiguration list by client")
+	}
+
+	return addonsConfigurationList, nil
+}
+
+func (r *ReconcileAddonsConfiguration) deleteOrphanBundles(namespace string) error {
+	existingBundles, err := r.bundleStorage.FindAll(internal.Namespace(namespace))
+	if err != nil {
+		return exerr.Wrap(err, "while listing bundles from storage")
+	}
+	addonsList, err := r.addonsConfigurationList(namespace)
+	if err != nil {
+		return exerr.Wrap(err, "while listing bundles")
+	}
+	for _, b := range existingBundles {
+		if r.hasBundleDeleted(b, addonsList) {
+			r.log.Infof("- delete bundle %s from storage", b.Name)
+			err := r.bundleStorage.Remove(internal.ClusterWide, internal.BundleName(b.Name), b.Version)
```

Comment:
It is `AddonsConfigurationController` therefore bundle with namespace should be removed